src/shogun/features/SNPFeatures.cpp

```diff
@@ -401,8 +401,9 @@ SGMatrix<float64_t> SNPFeatures::get_histogram(bool normalize)
 		}
 	}
 	SG_FREE(h_normalizer);
-
-	return SGMatrix<float64_t>(h, nsym, string_length/2);
+	SGMatrix<float64_t>hMatrix(h, nsym, string_length/2,false);
+	SG_FREE(h);
```

Comment:
why this free?
maybe it would be good to think about what actually would be happening with h. you can play with a debugger and see what happens line by line

Reply:
> @Khalifa1997 huh some crazy stuff happened with your branch. i guess you were rebasing something? in general would be great if you'd google around how to rebase a branch over another branch and some other 101 git operations. there are really tons of good material about this out there.
> 
> until you get of hold of that maybe just open a PR per commit/fix
Yup had plenty of problems with it, but yea will prob check some tutorials out




> why this free?
> maybe it would be good to think about what actually would be happening with h. you can play with a debugger and see what happens line by line


Okay will do, thanks!